public/src/components/channelManagement/epicTests/epicTestVariantsSplitEditor.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { ReactElement, useEffect } from 'react';
+import {
+  FormControl,
+  makeStyles,
+  Radio,
+  RadioGroup,
+  FormControlLabel,
+  TextField,
+  Theme,
+} from '@material-ui/core';
+import { ControlProportionSettings, EpicVariant } from './epicTestsForm';
+import { EMPTY_ERROR_HELPER_TEXT } from '../helpers/validation';
+import { useForm } from 'react-hook-form';
+
+const hasControl = (variants: EpicVariant[]): boolean =>
+  !!variants.find(v => v.name.toLowerCase() === 'control');
+
+const randomMvt = (): number => Math.round(Math.random() * 1000000);
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
+  variantsContainer: {
+    display: 'flex',
+    margin: '15px 30px 0 30px',
+    '& > * + *': {
+      marginLeft: spacing(3),
+    },
+  },
+  variantField: {
+    backgroundColor: '#EEEEEE',
+  },
+}));
+
+interface FormState {
+  percentage?: number;
+}
+
+interface EpicTestVariantsSplitEditorProps {
+  variants: EpicVariant[];
+  controlProportionSettings?: ControlProportionSettings;
+  onControlProportionSettingsChange: (
+    controlProportionSettings?: ControlProportionSettings,
+  ) => void;
+  onValidationChange: (isValid: boolean) => void;
+  isDisabled: boolean;
+}
+
+const EpicTestVariantsSplitEditor: React.FC<EpicTestVariantsSplitEditorProps> = ({
+  variants,
+  controlProportionSettings,
+  onControlProportionSettingsChange,
+  onValidationChange,
+  isDisabled,
+}: EpicTestVariantsSplitEditorProps) => {
+  const classes = useStyles();
+
+  const defaultValues = {
+    percentage: controlProportionSettings ? controlProportionSettings.proportion * 100 : undefined,
+  };
+
+  const { register, errors, handleSubmit, trigger, getValues, reset } = useForm<FormState>({
+    mode: 'onChange',
+    defaultValues,
+  });
+
```

Comment:
the `useEffects` below are not very intuitive, I'm finding `useForm` quite fiddly